### PR TITLE
Add reset button

### LINKED ISSUE #220: Unpublish portal button in the menu

### PR DESCRIPTION
**Pull request by RupengLiu**
Close #220
image

image

### CODE REVIEW

File: src/components/content/content.html [hunk 1]
```diff
@@ -4,5 +4,9 @@
             <i class="paperbits-icon paperbits-send"></i>
             Publish website
         </button>
+        <button class="btn" data-bind="click: unpublish">
+            <i class="paperbits-icon paperbits-simple-remove"></i>
+            Unpublish website
```
Comment (ygrik) [inline src/components/content/content.html]: maybe **Reset website/content** or **Website/Content reset**? Ask Mike what is better.

File: src/components/users/user-signup/ko/runtime/user-signup.html [hunk 2]
```diff
@@ -46,15 +46,6 @@
...
 
-            <!-- ko ifnot: isConsentRequired -->
-            <div class="row">
-                <p>
-                    Please review Terms of Use.
-                    <a href="#" class="btn-link" data-bind="click: toggleRegistrationTerms, text: showHideLabel"></a>
-                </p>
-            </div>
-            <!-- /ko -->
-
```
Comment (ygrik) [inline src/components/users/user-signup/ko/runtime/user-signup.html]: why did you remove this? Customers can set terms of use content but do not set that consent required for it.

File: src/components/users/user-signup/ko/runtime/user-signup.ts [hunk 3]
```diff
@@ -43,7 +43,7 @@ export class UserSignup {
         this.lastName = ko.observable("");
         this.isConsentRequired = ko.observable(false);
         this.consented = ko.observable(false);
-        this.showTerms = ko.observable();
+        this.showTerms = ko.observable(true);
```
Comment (ygrik) [inline src/components/users/user-signup/ko/runtime/user-signup.ts]: you should use toggleRegistrationTerms to avoid unsynchronized state

File: src/components/users/user-signup/ko/runtime/user-signup.ts [hunk 4]
```diff
@@ -92,7 +92,9 @@ export class UserSignup {
         }
         
         if (this.termsOfUse() && this.termsEnabled()) {
-            this.consented.extend(<any>{ equal: { params: true, message: "You must agree to registration terms." } });
+            if (this.isConsentRequired()) {
+                this.consented.extend(<any>{ equal: { params: true, message: "You must agree to registration terms." } });
+            }
             this.showHideLabel("Show");
```
Comment (ygrik) [inline src/components/users/user-signup/ko/runtime/user-signup.ts]: you should use toggleRegistrationTerms to avoid unsynchronized state

File: src/services/provisioningService.ts [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    ) { }
+
+    private async fetchData(url: string): Promise<Object> {
+        const response = await this.httpClient.send({ url: url, method: "GET" });
+        return response.toObject();
+    }
+
+    public async provision(): Promise<void> {
+        const settings = await this.settingsProvider.getSettings();
+        let managementApiUrl = settings["managementApiUrl"];
```
Comment (ygrik) [inline src/services/provisioningService.ts]: use the setting from Constants for "managementApiUrl"

File: src/services/provisioningService.ts [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+        if (!managementApiUrl) {
+            throw new Error(`Management API URL ("managementApiUrl") setting is missing in configuration file.`);
+        }
+        managementApiUrl = Utils.ensureUrlArmified(managementApiUrl);
+        const managementApiVersion = settings["managementApiVersion"];
```
Comment (ygrik) [inline src/services/provisioningService.ts]: use the setting from Constants for "managementApiVersion"

File: src/services/provisioningService.ts [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+        return response.toObject();
+    }
+
+    public async provision(): Promise<void> {
+        const settings = await this.settingsProvider.getSettings();
+        let managementApiUrl = settings["managementApiUrl"];
+        if (!managementApiUrl) {
+            throw new Error(`Management API URL ("managementApiUrl") setting is missing in configuration file.`);
+        }
+        managementApiUrl = Utils.ensureUrlArmified(managementApiUrl);
```
Comment (ygrik) [inline src/services/provisioningService.ts]: why are you doing this **Utils.ensureUrlArmified(managementApiUrl)**?

File: src/services/provisioningService.ts [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+
+    private async cleanupContent(): Promise<void> {
+        const settings = await this.settingsProvider.getSettings();
+        let managementApiUrl = settings["managementApiUrl"];
+        if (!managementApiUrl) {
+            throw new Error(`Management API URL ("managementApiUrl") setting is missing in configuration file.`);
+        }
+        managementApiUrl = Utils.ensureUrlArmified(managementApiUrl);
+        const url = `${managementApiUrl}/contentTypes?api-version=2018-06-01-preview`;
+        const accessToken = this.authenticator.getAccessToken();
```
Comment (ygrik) [inline src/services/provisioningService.ts]: please avoid this code duplication

File: src/services/provisioningService.ts [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+        const request: HttpRequest = {
+            url: url,
+            method: "GET",
+            headers: [
+                { name: "If-Match", value: "*" },
+                { name: "Content-Type", value: "application/json" },
+                { name: "Authorization", value: accessToken }
+            ],
+        };
+        const response = await this.sendRequest(request);
```
Comment (ygrik) [inline src/services/provisioningService.ts]: please wrap all http requests with try and catch

File: src/services/provisioningService.ts [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+                const itemReq: HttpRequest = {
+                    url: `${managementApiUrl}${item["id"]}?api-version=2018-06-01-preview`,
+                    method: "DELETE", 
+                    headers: [
+                        { name: "If-Match", value: "*" },
+                        { name: "Content-Type", value: "application/json" },
+                        { name: "Authorization", value: accessToken }
+                    ],
+                };
+                const response = await this.httpClient.send(itemReq);
```
Comment (ygrik) [inline src/services/provisioningService.ts]: wrap http requests with try and catch
and use existing function **sendRequest**

**Comment by mikebudzynski** [comment]
@RupengLiu 

1. The button should be "Reset content" not "Unpublish the portal"
1. The "Confirm reset" button should have margins and the label should be "Reset content"
1. @azaslonov is there another button style we can use for "Confirm reset"?
